### PR TITLE
crusher changes sept19

### PR DESCRIPTION
**Pull request by oksanaguba**
Needed for Crusher.

### CODE REVIEW

**Review by bartgol — CHANGES_REQUESTED** [review]
I think the version of kokkos you are using does not include stuff that we need in e3sm. You should branch off of origin/kokkos-3.6-with-clang-fix (what's currently used in ekat master), and merge the upstream kokkos version you need into your branch.

**Comment by oksanaguba** [comment]
@bartgol i am not sure what you mean. are we talking about ekat kokkos? then this branch points to June 10th commit (roughly needed for hip), while scream master version of ekat kokkos is from Feb 24, too old.

**Comment by bartgol** [comment]
> @bartgol i am not sure what you mean. are we talking about ekat kokkos? then this branch points to June 10th commit (roughly needed for hip), while scream master version of ekat kokkos is from Feb 24, too old.

I'm looking at the version of kokkos in ekat's master, and see a lot of commits that are not in the version of kokkos in your branch. For instance, this are the last few commits in the current master's kokkos version:

```
* 569fa3ac8 - (8 months ago) Disambiguate fujitsu-clang and clang compilers - Azamat Mametjanov (HEAD, origin/kokkos-3.6-with-clang-fix)
*   b52f8c835 - (6 months ago) Merge pull request #4827 from ndellingwood/release-3.6.00 - Christian Trott (tag: 3.6.00)
|\  
| * f2a05d316 - (6 months ago) Update master_history for Kokkos 3.6.00 - Nathan Ellingwood
| * 6ea708ffd - (6 months ago) Merge pull request #4967 from ndellingwood/rc-3.6.0-cherrypick-4966 - Damien L-G
| * faa8937ea - (6 months ago) Merge pull request #4944 from ndellingwood/rc-3.6.0-cherrypick-4943 - Damien L-G
| * 0a9e3b6eb - (6 months ago) Merge pull request #4934 from Rombur/fix_deep_copy_2 - Christian Trott
| * 66217f206 - (7 months ago) Update master_history for Kokkos 3.6.00 - Nathan Ellingwood
| * 1e9132944 - (7 months ago) Merge pull request #4866 from ndellingwood/rc-3.6.0-4852 - Christian Trott
| * 6b8493786 - (7 months ago) Merge pull request #4869 from crtrott/fix-scratch-locks - Christian Trott
| * 133de5679 - (7 months ago) Merge pull request #4851 from ndellingwood/rc-3.6.0-cherrypick-4841 - Damien L-G
| * 32736d35d - (7 months ago) Merge pull request #4865 from Rombur/fix_4850 - Christian Trott
| * 44e1b8487 - (8 months ago) Merge pull request #4830 from ndellingwood/rc-3.6.0-pthreadfix - Christian Trott
| * 2478f3b96 - (8 months ago) Merge branch 'release-candidate-3.6.0' for 3.6.00 - Nathan Ellingwood
|/| 
| *   c838bdcf4 - (8 months ago) Merge pull request #4807 from ndellingwood/rc-3.6.0-changelog - Damien L-G
```
Most of these are not in the version of kokkos in your branch, and I'm worried we might be messing up something here. Also, besides all the commits from the kokkos devs, we also want to keep the last commit, by Azamat, which fixes something that caused issues with our sems modules.

What I was suggesting is to create a new branch off of `569fa3ac8` (current version in ekat/master), and merge in whatever branch you are using from the upstream kokkos repo.

Also, if you want to upgrade kokkos, I would recommend to stick with their releases, rather than a random commit half way through. I would prefer picking their 3.7.00 tag. I checked the history of that tag, and it includes the last few commits of the version you are using.

**Comment by ambrad** [comment]
@bartgol are you opposed to our grabbing 3.7, say, and then cherry-picking our special commits onto that and making a tag of the result?

**Comment by bartgol** [comment]
> @bartgol are you opposed to our grabbing 3.7, say, and then cherry-picking our special commits onto that and making a tag of the result?

That's fine. I don't care how commits are organized in the tree, so long as we get the ones we added in our fork (there should be no more than 4 or 5).

File: cmake/machine-files/kokkos/hip.cmake [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Settings used when HIP is the Kokkos backend
+
+#is this needed?
+set(Kokkos_ENABLE_AGGRESSIVE_VECTORIZATION FALSE CACHE BOOL "")
```
Comment (bartgol) [inline cmake/machine-files/kokkos/hip.cmake:4]: Grepping the kokkos repo suggests this only matters for OpenMP and Threads backends.

**Review by bartgol — APPROVED** [review]

**Comment by ambrad** [comment]
@oksanaguba @bartgol are you ready to merge this? If so, it would be good to have this in before https://github.com/E3SM-Project/scream/pull/1797. Then that PR should have the ekat submodule updated to point to EKAT's updated master.

**Comment by oksanaguba** [comment]
I am ready, not sure what's required wrt testing for ekat.

**Comment by oksanaguba** [comment]
Since AT tested it, can this be merged? Should i press the green button?

**Comment by bartgol** [comment]
Ok, let's merge.

**Pull request merged**